### PR TITLE
Fixes the parsing of text formatting

### LINKED ISSUE #71: Minecraft TextFormatting not properly converted to Flint ChatColor
**Description**
When translating the Minecraft TextFormatting to Flint ChatColors when parsing a PlayerTeam, it just uses the name to get the Flint ChatColor.
https://github.com/FlintMC/FlintMC/blob/82a08750a62f40b1476360b9a4ad963ca310e64d/mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/world/scoreboard/VersionedScoreboardInterceptor.java#L141
But the TextFormatting can be "ITALIC" for example too, which is not included in the Flint ChatColor. This leads to a NullPointerException and the PlayerTeam parsing fails.

**To Reproduce the Bug**
Join on GommeHD.net for example

**Expected behavior**
The colors should be translated correctly or there should be decent error handling

**Log Output**
```
java.lang.NullPointerException: null
	at net.flintmc.mcapi.v1_15_2.world.scoreboard.listener.VersionedPlayerTeamChangeListener.lambda$changeColor$3(VersionedPlayerTeamChangeListener.java:90) ~[VersionedPlayerTeamChangeListener.class:?]
	at java.util.Optional.ifPresent(Optional.java:159) ~[?:1.8.0_144]
	at net.flintmc.mcapi.v1_15_2.world.scoreboard.listener.VersionedPlayerTeamChangeListener.changeColor(VersionedPlayerTeamChangeListener.java:88) ~[VersionedPlayerTeamChangeListener.class:?]
	at net.flintmc.mcapi.internal.world.scoreboard.score.DefaultPlayerTeam.setColor(DefaultPlayerTeam.java:137) ~[DefaultPlayerTeam.class:?]
	at net.flintmc.mcapi.v1_15_2.world.scoreboard.VersionedScoreboardInterceptor.hookHandleTeams(VersionedScoreboardInterceptor.java:141) ~[VersionedScoreboardInterceptor.class:?]
	at GeneratedMethodInjector_125_1f1ae294b45945eba506ea211023e950.notifyHook(GeneratedMethodInjector_125_1f1ae294b45945eba506ea211023e950.java) ~[?:?]
	at net.minecraft.client.network.play.ClientPlayNetHandler.handleTeams(ClientPlayNetHandler.java:1948) ~[ClientPlayNetHandler.class:?]
```

### PR DESCRIPTION
**Pull request by R0bbyYT**
Closes #71

### CODE REVIEW

File: mcapi/src/main/java/net/flintmc/mcapi/world/scoreboad/team/Team.java [hunk 1]
```diff
@@ -56,10 +58,26 @@ default boolean isSameTeam(Team other) {
...
+   * @return The optional team color.
+   */
+  Optional<ChatColor> getOptionalTeamColor();
+
+  /**
+   * Retrieves an optional chat format of this team.
+   *
+   * @return The optional team chat format.
+   */
+  Optional<ChatFormat> getOptionalChatFormat();
```
Comment (einrobin) [inline mcapi/src/main/java/net/flintmc/mcapi/world/scoreboad/team/Team.java]: Maybe remove the "Optional" from the name of these methods (and rename getOptionalTeamColor to getColor)?

File: mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/world/scoreboard/VersionedScoreboardInterceptor.java [hunk 2]
```diff
@@ -126,7 +130,8 @@ public void hookHandleDisplayObjective(@Named("args") Object[] args) {
   @Hook(
       className = "net.minecraft.client.network.play.ClientPlayNetHandler",
       methodName = "handleTeams",
-      parameters = {@Type(reference = STeamsPacket.class)})
+      parameters = {@Type(reference = STeamsPacket.class)},
+  version = "1.15.2")
```
Comment (einrobin) [inline mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/world/scoreboard/VersionedScoreboardInterceptor.java]: Apply autoformat

File: mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/world/scoreboard/listener/VersionedPlayerTeamChangeListener.java [hunk 3]
```diff
@@ -78,13 +78,18 @@ public void changeSuffix(PlayerTeam playerTeam, ChatComponent suffix) {
         (ITextComponent) this.componentMapper.toMinecraft(suffix)));
   }
 
+
+  @Override
```
Comment (einrobin) [inline mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/world/scoreboard/listener/VersionedPlayerTeamChangeListener.java:82]: Missing inheritDoc

File: mcapi/src/v1_16_5/java/net/flintmc/mcapi/v1_16_5/world/scoreboard/listener/VersionedPlayerTeamChangeListener.java [hunk 4]
```diff
@@ -78,13 +78,17 @@ public void changeSuffix(PlayerTeam playerTeam, ChatComponent suffix) {
         (ITextComponent) this.componentMapper.toMinecraft(suffix)));
   }
 
+  @Override
```
Comment (einrobin) [inline mcapi/src/v1_16_5/java/net/flintmc/mcapi/v1_16_5/world/scoreboard/listener/VersionedPlayerTeamChangeListener.java:81]: Missing inheritDoc

File: mcapi/src/internal/java/net/flintmc/mcapi/internal/world/scoreboard/score/DefaultPlayerTeam.java [hunk 5]
```diff
@@ -150,7 +154,13 @@ public void setFriendlyFlags(int flags) {
   @Override
   public void setColor(ChatColor color) {
     this.chatColor = color;
-    this.playerTeamChangeListener.changeColor(this, color);
+    this.playerTeamChangeListener.changeColor(this, color.getLowerName());
+  }
+
+  @Override
```
Comment (einrobin) [inline mcapi/src/internal/java/net/flintmc/mcapi/internal/world/scoreboard/score/DefaultPlayerTeam.java]: Missing inheritDoc

**Review by einrobin — CHANGES_REQUESTED** [review]

**Comment by juliarn** [comment]
What we thought about too is merging the ChatColor and ChatFormat enums together like it's done in Bukkit for example. Then we would not need a string and two getters for getting either the ChatFormat or ChatColor. In the most cases, the user propably does not know if it's a color or a format, so this would not only need weird handling in Flint but also weird handling from the user.

File: mcapi/src/internal/java/net/flintmc/mcapi/internal/world/scoreboard/DefaultScoreboard.java [hunk 6]
```diff
@@ -283,6 +283,7 @@ public void removePlayerFromTeam(String username, PlayerTeam team) {
   public boolean removePlayerFromTeams(String username) {
     PlayerTeam playerTeam = this.getPlayerTeam(username);
 
+    System.out.println("Removing... " + username);
```
Comment (juliarn) [inline mcapi/src/internal/java/net/flintmc/mcapi/internal/world/scoreboard/DefaultScoreboard.java]: This should probably be removed

**Review by juliarn — CHANGES_REQUESTED** [review]

File: mcapi/src/main/java/net/flintmc/mcapi/world/scoreboad/team/Team.java (hunk 1 above)
Comment (Janrupf) [reply mcapi/src/main/java/net/flintmc/mcapi/world/scoreboad/team/Team.java]: I also wonder if `Optional<ChatFormat>` makes sense, or if it should just return null.

**Review by juliarn — DISMISSED** [review]

**Review by juliarn — DISMISSED** [review]

**Review by einrobin — DISMISSED** [review]

**Review by juliarn — APPROVED** [review]

**Pull request merged**